stdlib/Statistics/src/Statistics.jl

```diff
@@ -645,9 +645,9 @@ function corm(x::AbstractVector, mx, y::AbstractVector, my)
 
     @inbounds begin
         # Initialize the accumulators
-        xx = zero(sqrt(abs2(x[1])))
-        yy = zero(sqrt(abs2(y[1])))
-        xy = zero(x[1] * y[1]')
+        xx = zero(typeof(sqrt(abs2(one(eltype(x))))))
```

Comment:
`typeof` isn't needed, right?

Reply:
No, it's not. Thanks for pointing out. I'll leave it out.

Reply:
Oh, and I wonder whether this might break things like `cor(Any[1, 2], Any[1, 2])`. Probably better do `one(x[1])` and so on so that the type of the first element is used when the element type is abstract. A test for that would be nice BTW.

Reply:
Yes, I was wondering if there is an issue with (in)homogeneity.